src/animal/animal.controller.ts

```diff
@@ -21,27 +22,37 @@ import {
   ApiResponse,
   ApiTags,
 } from '@nestjs/swagger';
-import { AnimalService } from './animal.service';
-import { fromAnimalModelToDto, fromAnimalMutateDtoToModel, fromAnimalsModelToDtos } from '../mappers/animal.mapper';
-import { AnimalDto } from 'src/dto/read.animal.dto';
-import { IncreaseAgeDto, IncreaseWeightDto, MutateAnimalDto } from 'src/dto/mutate.animal.dto';
-import { ANIMALS_API, APPLICATION_JSON_MEDIA_TYPE } from 'src/utils/constants';
+import { ANIMALS_API, APPLICATION_JSON_MEDIA_TYPE } from 'src/shared/constants';
+import {
+  fromAnimalsModelToDtos,
+  fromAnimalModelToDto,
+  fromAnimalMutateDtoToModel,
+} from 'src/shared/mapper/animal.mapper';
+import { AnimalDto, IncreaseAgeDto, IncreaseWeightDto } from 'src/shared/dto/animal/animal.dto';
+import { AnimalCreateDto } from 'src/shared/dto/animal/create-animal-dto';
+import { IAnimalService } from './interface/animal.service.inteface';
+import { ANIMAL_SERVICE_KEY } from './animal.providers';
 
 @ApiBearerAuth()
 @ApiTags(ANIMALS_API)
 @Controller(ANIMALS_API)
 export class AnimalController {
-  constructor(private service: AnimalService) {}
+  constructor(@Inject(ANIMAL_SERVICE_KEY) private service: IAnimalService) {}
 
   @Get()
+  @ApiOperation({ summary: 'Get all animals' })
   @ApiResponse({ description: 'Retrieved list of animals successfully.', type: AnimalDto, isArray: true })
+  @ApiProduces(APPLICATION_JSON_MEDIA_TYPE)
   async getAllAnimal(): Promise<AnimalDto[]> {
     return await this.service.getAll().then((result) => fromAnimalsModelToDtos(result));
   }
 
   @Get(':id')
   @ApiParam({ name: 'id', description: 'ID of the animal' })
+  @ApiOperation({ summary: 'Get animal by ID' })
   @ApiResponse({ description: 'Retrieved animal successfully.', type: AnimalDto })
+  @ApiProduces(APPLICATION_JSON_MEDIA_TYPE)
+  @ApiBadRequestResponse({ status: 404, description: 'Animal not found' })
   async getAnimalById(@Param('id') id: number): Promise<AnimalDto> {
     const value = await this.service.get(id);
```

Comment:
Use more meaningful variable names. `value` doesn't mean anything. For example use `animal` or `foundedAnimal`